js/widgets/forms/textinput.js

```diff
@@ -38,7 +38,7 @@ return $.widget( "mobile.textinput", {
 	initSelector: "input[type='text']," +
 		"input[type='search']," +
 		":jqmData(type='search')," +
-		"input[type='number']," +
+		"input[type='number']:not(:jqmData(type='range'))," +
```

Comment:
@arschmitz I'm actually a bit worried about the performance of this ```:not()``` - maybe I should short-circuit out of _create() if ```this.element.is( ":jqmData(type='range'))``` instead - might be faster.